core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1995,7 +2037,9 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(sortedBlocks.sameElements(decomManager.shufflesToMigrate.asScala.map(_._1)))
   }
 
-  class MockBlockTransferService(val maxFailures: Int) extends BlockTransferService {
+  class MockBlockTransferService(
+      val maxFailures: Int,
+      hostname: String = "MockBlockTransferServiceHost") extends BlockTransferService {
```

Comment:
nit: `override val hostName: ....`

So you don't need the `override def hostName: String = { hostname }` below..

Reply:
It didn't work. I think its probably because the hostname is a abstract method in `BlockTransferService`. Let me know if thats not the case.

Reply:
I mean:
```suggestion
      override val hostName: String = "MockBlockTransferServiceHost") extends BlockTransferService {
```

I just tried it and passed all tests.